pkg/env/msiauthorizer.go

```diff
@@ -10,6 +10,8 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/jongio/azidext/go/azidext"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/azcore"
```

Comment:
can we use sdk's azcore directly?

Reply:
I believe so.  I was thinking that I should follow the same structure, mainly so that the interfaces can be mocked.  Let me investigate that this is still needed now that azcore has `fake` and it's going to be GAed soon if not already.

Reply:
azcore Fake is there, but it's not as flexible as a mock.  It only allows simulating an error while fixing the content of the token with 24 hour expiration value.  

That said, I'm changing this to sdk's azcore anyway because there's no need to mock it yet.  If needed, it can be switch to mock easily later.